modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -941,13 +942,16 @@ public void preprocessSwagger(Swagger swagger) {
             }
             for (Operation operation : path.getOperations()) {
                 boolean hasFormParameters = false;
+                boolean hasBodyParameters = false;
                 for (Parameter parameter : operation.getParameters()) {
                     if (parameter instanceof FormParameter) {
                         hasFormParameters = true;
                     }
+                    if (parameter instanceof BodyParameter) {
+                        hasBodyParameters = true;
+                    }
                 }
-              //only add content-Type if its no a GET-Method
-                if(path.getGet() != null || ! operation.equals(path.getGet())){
+                if (hasBodyParameters){
```

Comment:
@bmordue shall we keep the logic `path.getGet() != null || ! operation.equals(path.getGet())` ?

For POST using HTTP form (not JSON/XML string), we also need the content-type right?

Reply:
Thanks, I've made sure the default content-type for form data gets applied too.

I don't think the original logic was correct: any path that includes a GET operation would have a default content-type applied, which doesn't seem right. I've added some tests that demonstrate this (ac799875265027d330e10a0efd485b9b0dfb3452).